src/zenml/utils/code_utils.py

```diff
@@ -168,55 +169,69 @@ def write_archive(
             )
 
 
-def upload_code_if_necessary(code_archive: CodeArchive) -> str:
-    """Upload code to the artifact store if necessary.
-
-    This function computes a hash of the code to be uploaded, and if an archive
-    with the same hash already exists it will not re-upload but instead return
-    the path to the existing archive.
+def zip_and_hash_code(code_archive: CodeArchive) -> Tuple[str, str]:
+    """Zip the code archive and compute its hash.
 
     Args:
-        code_archive: The code archive to upload.
+        code_archive: The code archive to zip and hash.
 
     Returns:
-        The path where to archived code is uploaded.
+        A tuple containing the temporary file path and the computed hash.
     """
-    artifact_store = Client().active_stack.artifact_store
-
     with tempfile.NamedTemporaryFile(
         mode="w+b", delete=False, suffix=".tar.gz"
     ) as f:
         code_archive.write_archive(f)
 
         hash_ = hashlib.sha1()  # nosec
 
+        f.seek(0)
```

Comment:
Was this a bug with the hash computation before or why is this `seek` needed?

Reply:
it was suggested by claude auto-complete

Reply:
That doesn't mean you shouldn't think about it or verify that it makes sense. In this case, the seeking already happens at the end of `code_archive.write_archive()` which means it's not necessary here.